libs/ngxtension/call-apply/src/call-apply.ts

```diff
@@ -0,0 +1,44 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+const error_this = function () {
+	throw new Error(
+		`DON'T USE this INSIDE A FUNCTION CALLED BY | call OR | apply IT MUST BE A PURE FUNCTION!`
+	);
+};
+const NOTHIS = !('Proxy' in window)
+	? Object.seal({})
+	: new Proxy(
+			{},
+			{
+				get: error_this,
+				set: error_this,
+				deleteProperty: error_this,
+				has: error_this,
+			}
+	  );
+
+@Pipe({
+	name: 'call',
+	pure: true,
+	standalone: true,
+})
+export class CallPipe implements PipeTransform {
+	transform(value: any, args?: Function): any {
+		if (typeof args !== 'function')
+			throw new TypeError('You must pass a PURE funciton to | call');
+		return args?.call(NOTHIS, value);
+	}
+}
+
+@Pipe({
+	name: 'apply',
+	pure: true,
+	standalone: true,
+})
+export class ApplyPipe implements PipeTransform {
+	transform(fn: Function, ...args: any[]): any {
```

Comment:
```suggestion
	transform<TFunction extends (...args: any[]) => any>(fn: TFunction, ...args: Parameters<TFunction>): ReturnType<TFunction> {
```

Reply:
Totally agree that this is a more strict type for the function and maybe enable TS inference, but II think that even adding this Angular will not "strict type-check" the pipes on template... I'll try and let's you know.

Reply:
And instead this will solve my NON strict-type check, great catch @nartc  
I tried in [stackblitz](https://stackblitz.com/edit/stackblitz-starters-grvarf?file=src%2Fcall-apply.ts,src%2Fmain.ts) your typed FN version emit an error (at build-time) if apply is called with wrong types... 
Don't see any "~swigel~" on template (maybe is the LanguageService implementation of stackblitz) or simply not supported case for Angular LSP, but at least we have more safe strict type-check!  

Do you want that I change even the Call sign allowing only function with 0-1 (param?: any) or even better `transform<T = any, R = any>(value: T, args?: (param?:T)=>R): R {`

Reply:
And then I revisit the tests and docs to remove the "NON strict" sample ;-)